doc/development/deprecations.rst

```diff
@@ -45,6 +45,12 @@ Pending deprecations
         some_qfunc(params)
         return qml.expval(Hamiltonian)
 
+* `Operator.__eq__` and `Operator.__hash__` will now raise a warning to indicate upcoming changes
+  to their behaviour.
```

Comment:
Since this page is about teaching people how to port their code, we can add a couple examples and motivations.  Something to get people excited for this change, like:
```
>>> qml.RX(1.0, wires=0) == qml.RX(1.0, wires=0)
True
>>> my_op_map = {qml.PauliX(0): "value"}
>>> my_op_map[qml.PauliX(0)]
"value"
```